uui-core/src/data/processing/views/LazyListView.ts

```diff
@@ -316,8 +326,13 @@ export class LazyListView<TItem, TId, TFilter = any> extends BaseListView<TItem,
             for (let n = 0; n < ids.length; n++) {
                 const id = ids[n];
                 const item = this.tree.getById(id);
-                const row = this.getRowProps(item, index, parents);
+                const hasNotIncludedParents = this.props.getParentId?.(item) && !parents.length;
```

Comment:
I'd rather always compute parents via the Tree. This shouldn't be slower, and we'll simplify the code by removing additional code paths.